server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java

```diff
@@ -141,6 +142,10 @@ public Writeable.Reader<RemoteConnectionInfo.ModeInfo> getReader() {
     }
 
     static ConnectionProfile buildConnectionProfile(String clusterAlias, Settings settings) {
+        final String transportProfile = REMOTE_CLUSTER_AUTHORIZATION.getConcreteSettingForNamespace(clusterAlias).exists(settings)
+            ? RemoteClusterPortSettings.REMOTE_CLUSTER_PROFILE
+            : TransportSettings.DEFAULT_PROFILE;
```

Comment:
The existence of the `authorization` setting is the way to differentiate between legacy and new remote clusters. For legacy remote clusters, we keep use the `default` profile. For new ones, we use `_remote_cluster`.